Automattic/jetpack
# VideoPress: Allow user to update poster selecting a frame from video

Implement the feature to allow the user select a poster image using a frame from the video.

Closes #26153 

### ⚠️ KNOWN BUG

If you reload the page, it won't use the new poster image, for some reason we're not getting fresh data. But if you navigate to library and open the attachment, the player will load based on fresh data.

#### Changes proposed in this Pull Request:

* Move `usePosterUpload` and `usePosterImage` hooks out of `videopress block`.
* Extract `VideoFrameSelector` from `videopress block`.
* Fix dropdown not closing in `VideoThumbnail`.
* Allow rendering node in `thumbnail` space at `VideoThumbnail`.
* Support videopress shared components in `Storybook`.
* Extract `PlayIcon` from `videopress block`.

#### Other information:


#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Start `VideoPress`
* Navigate to its` dashboard `.
* Click on `EditDetails` button.
* At `thumbnail`, click on `edit` button.
* Select the second option - `Select from `video`
* Use the range to select a frame.
* Click on `Select Frame`.
* And save changes.
* You should wait and see the new poster from frame.
* Confirm it from dev tools.

#### Demo


https://user-images.githubusercontent.com/1663717/191580266-4a8cfd43-9c44-414d-b17b-84c492343aa6.mp4
